config/clients/js/template/client.mustache

```diff
@@ -0,0 +1,536 @@
+  async readAuthorizationModels(options: ClientRequestOpts & PaginationOptions = {}): PromiseResult<ReadAuthorizationModelsResponse> {
+    return this.api.readAuthorizationModels(options.pageSize, options.continuationToken, options);
+  }
+
+  /**
+   * WriteAuthorizationModel - Create a new version of the authorization model
+   * @param {WriteAuthorizationModelRequest} body
+   * @param {ClientRequestOpts} [options]
+   * @param {object} [options.headers] - Custom headers to send alongside the request
+   * @param {object} [options.retryParams] - Override the retry parameters for this request
+   * @param {number} [options.retryParams.maxRetry] - Override the max number of retries on each API request
+   * @param {number} [options.retryParams.minWaitInMs] - Override the minimum wait before a retry is initiated
+   */
+  async writeAuthorizationModel(body: WriteAuthorizationModelRequest, options: ClientRequestOpts = {}): PromiseResult<WriteAuthorizationModelResponse> {
+    return this.api.writeAuthorizationModel(body, options);
+  }
+
+  /**
+   * ReadAuthorizationModel - Read the current authorization model
+   * @param {ClientRequestOptsWithAuthZModelId} [options]
+   * @param {string} [options.authorizationModelId] - Overrides the authorization model id in the configuration
+   * @param {object} [options.headers] - Custom headers to send alongside the request
+   * @param {object} [options.retryParams] - Override the retry parameters for this request
+   * @param {number} [options.retryParams.maxRetry] - Override the max number of retries on each API request
+   * @param {number} [options.retryParams.minWaitInMs] - Override the minimum wait before a retry is initiated
+   */
+  async readAuthorizationModel(options: ClientRequestOptsWithAuthZModelId = {}): PromiseResult<ReadAuthorizationModelResponse> {
+    const authorizationModelId = this.getAuthorizationModelId(options);
+    if (!authorizationModelId) {
+      throw new Error("authorization_model_id_required");
+    }
+    return this.api.readAuthorizationModel(authorizationModelId, options);
+  }
+
+  /**
+   * ReadLatestAuthorizationModel - Read the latest authorization model for the current store
+   * @param {ClientRequestOpts} [options]
+   * @param {object} [options.headers] - Custom headers to send alongside the request
+   * @param {object} [options.retryParams] - Override the retry parameters for this request
+   * @param {number} [options.retryParams.maxRetry] - Override the max number of retries on each API request
+   * @param {number} [options.retryParams.minWaitInMs] - Override the minimum wait before a retry is initiated
+   */
+  async readLatestAuthorizationModel(options: ClientRequestOpts = {}): PromiseResult<ReadAuthorizationModelResponse> {
+    const { headers = {} } = options;
+    setHeaderIfNotSet(headers, CLIENT_METHOD_HEADER, "ReadLatestAuthorizationModel");
+    const authorizationModelsResponse = await this.readAuthorizationModels({ ...options, headers });
```

Comment:
should we ensure options has pageSize = 1?

Reply:
Good idea!